### PR TITLE
newer tool versions caught an unused variable scenario

### PR DESCRIPTION
**Pull request by kaleb-himes**
# Description

Please describe the scope of the fix or feature addition.

Fixes zd#

# Testing

How did you test?

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: mcapi/crypto.c [hunk 1]
```diff
@@ -424,7 +424,7 @@ int CRYPT_TDES_CBC_Decrypt(CRYPT_TDES_CTX* tdes, unsigned char* out,
 int CRYPT_AES_KeySet(CRYPT_AES_CTX* aes, const unsigned char* key,
                      unsigned int keyLen, const unsigned char* iv, int dir)
 {
-    typedef char aes_test[sizeof(CRYPT_AES_CTX) >= sizeof(Aes) ? 1 : -1];
+    typedef char aes_test[sizeof(CRYPT_AES_CTX) >= sizeof(Aes) ? 1 : 0];
```
Comment (douzzer) [inline mcapi/crypto.c:427]: The `[-1]` here is deliberate, to derail compilation.  the code pattern is quite common:
```
$ egrep -c '\[[^]]*:[[:space:]]*-1[[:space:]]*\]' mcapi/*.c src/*.c | fgrep -v :0
mcapi/crypto.c:9
src/internal.c:7
src/sniffer.c:1
src/ssl.c:14
src/tls13.c:6
```

File: wolfcrypt/src/integer.c [hunk 2]
```diff
@@ -1751,6 +1755,10 @@ int s_mp_add (mp_int * a, mp_int * b, mp_int * c)
       *tmpc++ &= MP_MASK;
     }
 
+    /* ensure pointer still points to valid memory before proceeding */
+    if (!tmpc)
```
Comment (douzzer) [inline wolfcrypt/src/integer.c:1759]: Obviously this should not be necessary -- the nullness check on `c->dp` precludes `tmpc` being null.

If an analyzer is complaining here, that sounds like a false positive.

**Review by douzzer — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/integer.c (hunk 2 above)
Comment (kaleb-himes) [reply wolfcrypt/src/integer.c:1759]: If the pointer always pointed to the same address I would agree.
Line 1755 changes the address the pointer is pointing to.
I do not think it is a "bad thing" to ensure the new address is still a valid address after the address has been modified multiple times in the for loop.

Comment (kaleb-himes) [reply wolfcrypt/src/integer.c:1759]: Alternative: To consider this a false positive then we must consider both clang-10 and clang-12 to be broken (which both report on this item) and remove the test for this build that used to pass in favor of just not testing it. Let me know.

**Comment by kaleb-himes** [comment]
Closed after talking w/ @dgarske

**Pull request closed without merging**